externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -178,23 +174,29 @@ class ExecuteStatement(
           resultSaveThreshold,
           result)) {
         val sessionId = session.handle.identifier.toString
-        val savePath = session.sessionManager.getConf.get(OPERATION_RESULT_SAVE_TO_FILE_DIR)
-        saveFileName = Some(s"$savePath/$engineId/$sessionId/$statementId")
-        // Rename all col name to avoid duplicate columns
-        val colName = range(0, result.schema.size).map(x => "col" + x)
-
-        val codec = if (SPARK_ENGINE_RUNTIME_VERSION >= "3.2") "zstd" else "zlib"
-        // df.write will introduce an extra shuffle for the outermost limit, and hurt performance
-        if (resultMaxRows > 0) {
-          result.toDF(colName: _*).limit(resultMaxRows).write
-            .option("compression", codec).format("orc").save(saveFileName.get)
-        } else {
-          result.toDF(colName: _*).write
-            .option("compression", codec).format("orc").save(saveFileName.get)
+        saveFileName = {
+          val resultSaveToDir =
+            session.sessionManager.getConf.get(OPERATION_RESULT_SAVE_TO_FILE_DIR)
+          Some(s"$resultSaveToDir/$engineId/$sessionId/$statementId")
         }
-        info(s"Save result to $saveFileName")
+        Utils.addShutdownHook(() => cleanupSavedResultFile())
```

Comment:
After the engine exits, the entire directory will be cleaned, which is more efficient than cleaning files one by one.

If you want to clean up the query results quickly, I suggest deleting it directly in the close method.

https://github.com/apache/kyuubi/blob/4cceb1dbd72ec9d6ac0c9dc9a5b2b21f0600ba9b/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala#L94-L101

Reply:
Already did.

https://github.com/apache/kyuubi/blob/e9e2d189ba05c899fd85041e2a3bb7db4f5f009f/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala#L70-L78